features/welcome/actions.js

```diff
@@ -97,13 +116,29 @@ export function conferenceJoined(conference) {
 /**
  * Create an action for when the signaling connection has been lost.
  *
- * @param {string} message - Error message.
+ * @param {string} [message] - Error message.
  * @returns {{type: JITSI_CLIENT_DISCONNECTED, message: string}}
  */
 export function connectionDisconnected(message) {
-    return {
-        type: JITSI_CLIENT_DISCONNECTED,
-        message
+    return dispatch => {
+        dispatch(removeLocalParticipant());
+
+        return dispatch(removeLocalTracks())
+            .then(() => dispatch({ type: JITSI_CLIENT_DISCONNECTED, message }));
+    };
+}
+
+/**
+ * Create an action for when the signaling conference was left.
+ *
+ * @returns {{type: JITSI_CONFERENCE_LEFT, message: string}}
+ */
+export function conferenceLeft() {
+    return dispatch => {
+        dispatch(resetToolbar());
```

Comment:
That's coupling between the welcome and toolbar feature. The toolbar feature should observe the model and reset itself, not await another feature to tell it that it needs to reset.